### PR TITLE
vi: Translate Casting and Read/Write chapters from Std Trait session

### PR DESCRIPTION
**Pull request by vohoanglong0107**
A small one. The final PR for `Std Trait` session will include the `Default` and `Closures` chapter, along with the session's exercises.

### CODE REVIEW

File: po/vi.po [hunk 1]
```diff
@@ -6724,54 +6736,71 @@ msgid ""
...
 
 #: src/std-traits/read-and-write.md
 msgid "\"lines in slice: {}\""
-msgstr ""
+msgstr "\"số dòng trong mảng: {}\""
 
 #: src/std-traits/read-and-write.md
 msgid "\"lines in file: {}\""
-msgstr ""
+msgstr "\"số dòng trong file: {}\""
```
Comment (daivinhtran) [inline po/vi.po:6787]: số dòng is the number of lines. Let's say "những dòng"

File: po/vi.po [hunk 2]
```diff
@@ -6724,54 +6736,71 @@ msgid ""
...
 
 #: src/std-traits/read-and-write.md
 msgid "b\"foo\\nbar\\nbaz\\n\""
-msgstr ""
+msgstr "b\"foo\\nbar\\nbaz\\n\""
 
 #: src/std-traits/read-and-write.md
 msgid "\"lines in slice: {}\""
-msgstr ""
+msgstr "\"số dòng trong mảng: {}\""
```
Comment (daivinhtran) [inline po/vi.po:6783]: số dòng is the number of lines. Let's say "những dòng"

File: po/vi.po [hunk 3]
```diff
@@ -6724,54 +6736,71 @@ msgid ""
...
 msgstr ""
+"`as` hoạt động tương tự như static cast trong C++. Không khuyến khích sử "
+"dụng `as` trong những trường hợp dữ liệu có thể bị mất trong quá trình "
+"chuyển đổi, ít nhất cần phải có một comment giải thích rõ ràng lý do sử dụng "
+"`as`."
 
 #: src/std-traits/casting.md
 msgid "This is common in casting integers to `usize` for use as an index."
 msgstr ""
+"Một trường hợp ngoại lệ là khi chuyển đổi số nguyên sang `usize` để sử dụng "
```
Comment (daivinhtran) [inline po/vi.po:6764]: "common" isn't "ngoại lệ". should be "thông thường"?

**Review by daivinhtran — COMMENTED** [review]
Fanstatic translations, @vohoanglong0107 !

I have a few minor comments. Let me know once you fix them.

File: po/vi.po (hunk 3 above)
Comment (vohoanglong0107) [reply po/vi.po:6764]: I would like to combine this translation with the above paragraph, forming
```
Không khuyến khích sử
dụng `as` trong những trường hợp dữ liệu có thể bị mất trong quá trình
chuyển đổi, ít nhất cần phải có một comment giải thích rõ ràng lý do sử dụng
`as`.
Một trường hợp ngoại lệ là khi chuyển đổi số nguyên sang `usize` để sử dụng
như một index trong array.
```
Do you think this is a good idea?

File: po/vi.po (hunk 2 above)
Comment (vohoanglong0107) [reply po/vi.po:6783]: Hmmm, this line is used in this example:
```rust
    let slice: &[u8] = b"foo\nbar\nbaz\n";
    println!("lines in slice: {}", count_lines(slice));
```
Therefore I think `số dòng` would be a better translation, as we are counting the lines in slice here.

File: po/vi.po (hunk 1 above)
Comment (vohoanglong0107) [reply po/vi.po:6787]: This line is used in this example:
```rust
    let file = std::fs::File::open(std::env::current_exe()?)?;
    println!("lines in file: {}", count_lines(file));
```
I think `số dòng` would be a better translation than `những dòng` in this case.

File: po/vi.po (hunk 3 above)
Comment (vohoanglong0107) [reply po/vi.po]: I would like to combine this translation to the above paragraph, forming:
```
Đối với những trường hợp ép kiểu có
thể thất bại, nên sử dụng `TryFrom` và `TryInto` để xử lý lỗi khi ép kiểu
(khi một giá trị không hoàn toàn phù hợp với kiểu dữ liệu đích).
Một trường hợp ngoại lệ là khi chuyển đổi số nguyên sang `usize` để sử dụng
làm index trong array
```
Do you think this is a good idea?

File: po/vi.po (hunk 1 above)
Comment (vohoanglong0107) [reply po/vi.po:6784]: Hmmm, this line is used like this in the example:
```rust
    let slice: &[u8] = b"foo\nbar\nbaz\n";
    println!("lines in slice: {}", count_lines(slice));
```
In my opinion `số dòng` would make a better translation, as we are counting for lines in the example

File: po/vi.po (hunk 2 above)
Comment (vohoanglong0107) [reply po/vi.po:6780]: This line is used like this in the example:
```rust
    let file = std::fs::File::open(std::env::current_exe()?)?;
    println!("lines in file: {}", count_lines(file));
```
I think `số dòng` would be more fitting in the given context.

File: po/vi.po (hunk 1 above)
Comment (daivinhtran) [reply po/vi.po:6643]: SGTM

File: po/vi.po (hunk 3 above)
Comment (daivinhtran) [reply po/vi.po]: Yeah that's fine for me.

**Review by qu-ngx — APPROVED** [review]
SGTM! I think it looks good now!

**Pull request merged**